src/Send/MemoField.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {TextInput} from '../components'
+import {COLORS} from '../theme'
+import {useSend} from './Context/SendContext'
+
+const maxMemoValue = 256
+
+export const MemoField = () => {
+  const strings = useStrings()
+  const {memo, memoChanged} = useSend()
+
+  const showError = memo.length > maxMemoValue
+
+  return (
+    <View style={styles.container}>
+      <TextInput
+        value={memo}
+        onChangeText={(memo) => memoChanged(memo.trim())}
+        label={strings.label}
+        autoComplete={false}
+        testID="memoFieldInput"
+        errorText={showError ? 'error' : undefined}
+        noErrors={true}
+      />
+
+      <View style={styles.helper}>
+        <Text style={[styles.warning, showError && styles.error]}>{strings.warning}</Text>
+
+        <Text style={[styles.counter, showError && styles.error]}>{`${memo.length}/${maxMemoValue}`}</Text>
+      </View>
+    </View>
```

Comment:
`HelperText` component from `react-native-paper` is limited so I created a custom helper.